tests/test_entity_preprocessor.py

```diff
@@ -0,0 +1,45 @@
+import unittest
+
+from datasets import Dataset
+
+from bsmetadata.preprocessing_utils import EntityPreprocessor
+
+
+class TestEntityPreprocessor(unittest.TestCase):
+    def test_extract_entities(self):
+
+        my_dict = {
+            "id": [0, 1, 2],
+            "text": [
+                "Paris is the beautiful place to visit",
+                "Obama was the guest of honor at the conference.",
+                "Bieber performed at the concert last night",
```

Comment:
I would advise maybe to take an example where the entity is not at the beginning to check the `char_start_idx` and `char_end_idx.